numba/dummyarray.py

```diff
@@ -292,18 +292,77 @@ def reshape(self, *newshape, **kws):
 
         if self.is_c_contig or self.is_f_contig:
             if order == 'C':
-                newstrides = list(iter_strides_c_contig(self, newshape))
+                newstrides = list(iter_strides_c_contig(self, newdims))
             elif order == 'F':
-                newstrides = list(iter_strides_f_contig(self, newshape))
+                newstrides = list(iter_strides_f_contig(self, newdims))
             else:
                 raise AssertionError("unreachable")
-        elif newnd == 1:
-            s = [self.strides[i] for i in range(oldnd) if self.shape[i] > 1]
-            newstrides = (min(s) if len(s) else self.itemsize,)
         else:
-            raise NotImplementedError("reshape on non-contiguous array")
+            # Transliteration of numpy's `_attempt_nocopy_reshape`
+
+            # Remove axes with dimension 1 from the old array. They have no effect
+            # but would need special cases since their strides do not matter.
+            olddims, oldstrides = zip(*[
```

Comment:
The `zip(...)` here can yield nothing, leading to an unpack on empty sequence.  This is revealed in test: `numba.cuda.tests.cudadrv.test_cuda_array_slicing.CudaArraySlicing.test_prefix_select`

Reply:
Ah - pushed a fix for that, `cudadrv` & `cudapy` tests now pass